src/macros.rs

```diff
@@ -55,12 +55,17 @@ macro_rules! interchange {
                 static mut INTERCHANGES: [Option<$Name>; $N] = [NONE; $N];
                 static mut STATES: [u8; $N] = [0u8; $N];
                 unsafe {
-                    let mut cell: MaybeUninit<UnsafeCell<&'static mut Option<$Name>>> = MaybeUninit::uninit();
+                    // It's unnecessary to wrap interchange in Option<T> because of the state handling we are
+                    // already doing.  Since Option<T> must take atleast as much space as T, 
+                    // we transmute &mut Option<T> to &mut T.  There is another way to use MaybeUninit but it's not
+                    // yet stable.
+                    let underlying_data: &'static mut $Name = unsafe { core::mem::transmute::<_, _>(&mut INTERCHANGES[i]) };
```

Comment:
I don't think this transmute is valid. At least I don't see an argument for it to be so; you seem to be making assumptions on the basis of "it must fit"?

Also, what is the "another way" with MaybeUninit?

Reply:
I think this works and is less iffy:
```
#[derive(Clone, Debug, PartialEq)]    
pub enum $Name {    
    Request($REQUEST),    
    Response($RESPONSE),    
    None,    
}
(...)
static mut INTERCHANGES: [$Name; $N] = [$Name::None; $N];
(...)
cell.as_mut_ptr().write(UnsafeCell::new(&mut INTERCHANGES[i])); 
```